app/controllers/albums_controller.rb

```diff
@@ -0,0 +1,56 @@
+class AlbumsController < ApplicationController
+  before_action :get_album, only: [:show, :edit, :destroy, :upvote, :update]
+
+  def get_album
+    @album = Album.find(params[:id])
+  end
+
+  def index
+    @albums = Album.order(ranked: :desc)
+  end
+
+  def show
+  end
+
+  def new
+    @album = Album.new
+  end
+
+  def create
+    @album = Album.create(album_params)
+    if @album.save
+      redirect_to album_path(@album)
+    else
+      render "new"
+    end
+  end
+
+  def edit
+  end
+
+  def update
+    @album.update(album_params)
+    if @album.save
+      redirect_to album_path(@album)
+    else
+      render "edit"
+    end
+  end
+
+  def destroy
+    @album.destroy
+    redirect_to albums_path
+  end
+
+  def upvote
+    @album.ranked += 1
+    @album.save
+    redirect_to album_path(@album)
+  end
+
+  private
+
+  def album_params
+    params.require(:album).permit([:name, :artist, :description])
```

Comment:
I like that you did not include `:ranked` here -- this is the exact purpose of the strong params idea.

Because none of our routes intend for the user to directly modify the `ranked` attribute (even in upvote we just modify it ourselves), we should not permit that attribute to exist in our params. If we had allowed it, a malicious user could add that into the params hash directly and then Rails would be happy to let them set whatever value they wanted.